File: content/http-body-to-file/a2.nim
import asyncdispatch, httpclient, strformat
proc f1(n1, n2, n3: int64) {.async.} =
   echo &"total: {n1}, received: {n2}, speed: {n3 div 1000}k"
proc f2(s3: string) {.async.} =
   var o1 = newAsyncHttpClient()
   o1.onProgressChanged = f1
   await o1.downloadFile(s3, "100mb.file")
var s2 = "http://speedtest.lax.hivelocity.net/100mb.file"
waitFor f2(s2)

File: content/http-body-to-file/a2.php
<?php
$s1 = 'http://speedtest.lax.hivelocity.net/100mb.file';
$r1 = fopen('100mb.file', 'w');
$r2 = curl_init();
curl_exec($r2);
curl_setopt($r2, CURLOPT_FILE, $r1);
curl_setopt($r2, CURLOPT_NOPROGRESS, false);
